MetaMask/metamask-extension
# feat: Hook in Portfolio Entry Points

## **Description**

1. What is the reason for the change?
Portfolio has requested to add in some entry points into the extension, so users can easily navigate to the Portfolio to view/ manage their spending caps.
2. What is the improvement/solution?
This adds value for the users who would like to view/ manage their spending caps as well as their portfolio.

[![Open in GitHub Codespaces](https://github.com/codespaces/badge.svg)](https://codespaces.new/MetaMask/metamask-extension/pull/27607?quickstart=1)

## **Manual testing steps**

1. Connect an account
2. Go to this the assets page in the extension
3. Click on an asset
4. Under "Token details", there should be a category for all native and non-native token types for "Spending caps"
5. Next to Spending caps, check the there is a link that routes the user to the portfolio with the "spendingCaps" tab and the user's account address passed as query params
6. Check that the link redirects to Portfolio

## **Screenshots/Recordings**


### **Before**
<img width="359" alt="Screenshot 2024-10-03 at 10 45 22 AM" src="https://github.com/user-attachments/assets/783f8d0f-248f-4729-84cf-f100a7a2cb04">

### **After**
<img width="358" alt="Screenshot 2024-10-03 at 9 37 54 AM" src="https://github.com/user-attachments/assets/7dd2cb6f-3a1a-4245-a71b-e9186775ce87">
<img width="357" alt="Screenshot 2024-10-03 at 9 33 21 AM" src="https://github.com/user-attachments/assets/8b3b487b-0ef8-4a19-9407-95ef973d7002">
<img width="358" alt="Screenshot 2024-10-03 at 9 32 20 AM" src="https://github.com/user-attachments/assets/05618a7b-4d50-4871-a4b7-fbb860dcadf7">




## **Pre-merge author checklist**


## **Pre-merge reviewer checklist**
